crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -47,18 +58,22 @@ impl C8yEndPoint {
         url_get_id
     }
 
-    pub fn get_url_for_sw_list(&self) -> String {
+    pub fn get_url_for_sw_list(&self, device_id: Option<&str>) -> Result<String, C8yEndPointError> {
```

Comment:
Using `None ` for the main device is a trick that makes the code not crystal clear.

A solution could be to introduce a specific enum with `MainDevice` and `ChildDevice("xyz")` cases.

Reply:
refactored